include/util/concurrent_id_map.hpp

```diff
@@ -27,6 +27,23 @@ struct ConcurrentIDMap
     std::unordered_map<KeyType, ValueType, HashType> data;
     mutable UpgradableMutex mutex;
 
+    ConcurrentIDMap() = default;
+    ConcurrentIDMap(ConcurrentIDMap &&other)
+    {
+        ScopedWriterLock other_lock{other.mutex};
+        ScopedWriterLock lock{mutex};
+
+        data = std::move(other.data);
+    }
+    ConcurrentIDMap &operator=(ConcurrentIDMap &&other)
+    {
+        ScopedWriterLock other_lock{other.mutex};
+        ScopedWriterLock lock{mutex};
+
+        data = std::move(other.data);
+        return *this;
+    }
```

Comment:
to be on the safe side I would add a check for `this != &other` otherwise a deadlock occurs in the code above